libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -1439,11 +1439,20 @@ typedef __char32_t char32_t;
 #endif
 #endif
 
+// We want to hide defaulted SMF but those cannot be marked as __host__ __device__
+#ifndef _LIBCUDACXX_HIDE_FROM_ABI_DEFAULTED
+#  if _LIBCUDACXX_HIDE_FROM_ABI_PER_TU
+#    define _LIBCUDACXX_HIDE_FROM_ABI_DEFAULTED inline _CCCL_ATTRIBUTE_HIDDEN _LIBCUDACXX_INTERNAL_LINKAGE
+#  else
+#    define _LIBCUDACXX_HIDE_FROM_ABI_DEFAULTED inline _CCCL_ATTRIBUTE_HIDDEN _LIBCUDACXX_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
+#  endif
+#endif
+
 #ifndef _LIBCUDACXX_HIDE_FROM_ABI
 #  if _LIBCUDACXX_HIDE_FROM_ABI_PER_TU
-#    define _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HIDDEN _LIBCUDACXX_INTERNAL_LINKAGE
+#    define _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HIDE_FROM_ABI_DEFAULTED _LIBCUDACXX_HOST_DEVICE
 #  else
-#    define _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HIDDEN _LIBCUDACXX_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
+#    define _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HIDE_FROM_ABI_DEFAULTED _LIBCUDACXX_HOST_DEVICE
```

Comment:
suggestion: both branches are equivalent. If the point of keeping the branch is documentation, I'd suggest to just add a comment that unlike libc++ we do not exclude from explicit instantiation for `!_LIBCUDACXX_HIDE_FROM_ABI_PER_TU` and explain why.